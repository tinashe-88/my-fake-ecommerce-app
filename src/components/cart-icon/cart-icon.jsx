import React from 'react'
import { connect } from 'react-redux'
import { createStructuredSelector } from 'reselect'

import { toggleCartHidden } from '../../redux/cart/cart.actions'
import { selectCartItemsQuantity } from '../../redux/cart/cart.selectors'

import {
  CartIconContainer, 
  ShoppingIconContainer, 
  ItemCountContainer
} from './cart-icon.styles'

const CartIcon = ({ toggleCartHidden, itemQuantity }) => (
  <CartIconContainer onClick={toggleCartHidden}>
    <ShoppingIconContainer/>
    <ItemCountContainer>{itemQuantity}</ItemCountContainer>
  </CartIconContainer>
)

// Selector
const mapStateToDispatch = dispatch => ({
  toggleCartHidden: () => dispatch(toggleCartHidden())
})

const mapStateToProps = createStructuredSelector({
  itemQuantity: selectCartItemsQuantity
})

export default connect(mapStateToProps, mapStateToDispatch)(CartIcon)