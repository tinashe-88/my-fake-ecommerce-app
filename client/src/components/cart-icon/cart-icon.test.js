import React from 'react'
import { shallow } from 'enzyme'
import { CartIcon } from './cart-icon'

describe('CartIcon Component', () => {
  let wrapper
  let mockToggleCartHidden

  beforeEach(() => {
    mockToggleCartHidden = jest.fn()

    const mockProps = {
      itemQuantity: 0,
      toggleCartHidden: mockToggleCartHidden
    }

    wrapper = shallow(<CartIcon {...mockProps}/>)
  })

  it('Should render CartIcon component', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should call toggleCartHidden when icon is clicked', () => {
    wrapper.find('CartIconContainer').simulate('click')
    expect(mockToggleCartHidden).toHaveBeenCalled()
  })

  it('should render the itemQuantity as the text', () => {
    const itemQuantity = parseInt(wrapper.find('ItemCountContainer').text())
    expect(itemQuantity).toBe(0)
  })
})