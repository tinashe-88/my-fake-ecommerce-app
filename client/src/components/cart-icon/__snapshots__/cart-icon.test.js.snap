// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartIcon Component Should render CartIcon component 1`] = `
<CartIconContainer
  onClick={[MockFunction]}
>
  <Styled(Component) />
  <ItemCountContainer>
    0
  </ItemCountContainer>
</CartIconContainer>
`;
